lib/auth/join.go

```diff
@@ -93,10 +95,41 @@ func (a *Server) RegisterUsingToken(ctx context.Context, req *types.RegisterUsin
 	}
 
 	// perform common token checks
-	if err := a.checkTokenJoinRequestCommon(ctx, req); err != nil {
+	provisionToken, err := a.checkTokenJoinRequestCommon(ctx, req)
+	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	certs, err := a.generateCerts(ctx, provisionToken, req)
+	return certs, trace.Wrap(err)
+}
+
+func (a *Server) generateCerts(ctx context.Context, provisionToken types.ProvisionToken, req *types.RegisterUsingTokenRequest) (*proto.Certs, error) {
+	if req.Role == types.RoleBot {
+		// bots use this endpoint but get a user cert
+		// botResourceName must be set, enforced in CheckAndSetDefaults
+		botResourceName := provisionToken.GetBotName()
+		expires := a.GetClock().Now().Add(defaults.DefaultRenewableCertTTL)
```

Comment:
Why does this generate the cert with default cert TTL instead of the one the bot is configured with?

Reply:
It's somewhat arbitrary. We don't pass along a desired TTL for the initial certs via the token request here which complicates things. That said, in practice bots immediately attempt to renew certs after joining, and do request a user-configurable TTL then. So for however long these initial certs last, in practice they are disposed of within seconds.

That being said, the `DefaultRenewableCertTTL` is way too long, I'll set it to 1 hour.